### PR TITLE
fix: set go module to v2

### LINKED ISSUE #475: Please update module to be github.com/numtide/treefmt/v2
**Describe the bug**

[here](https://github.com/numtide/treefmt/blob/main/go.mod#L1)

When the project has versions 2.x and higher module has to end with corresponding /v2 and higher.
Otherwise Golang servers don't pick up project sources.

See details [here](https://go.dev/blog/v2-go-modules).

### PR DESCRIPTION
**Pull request by brianmcgee**
Closes #475

Signed-off-by: Brian McGee <brian@bmcgee.ie>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ To reformat the whole source tree, just type `treefmt` in any folder. This is a
 
 ## Installation
 
-You can install `treefmt` by downloading the binary. Find the binaries for different architectures [here](https://github.com/numtide/treefmt/releases).
+You can install `treefmt` by downloading the binary. Find the binaries for different architectures [here](https://github.com/numtide/treefmt/v2/releases).
```
Comment (jfly) [inline README.md:50]: Typo?

File: README.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ To explore the tool’s flags and options, type:
 $ treefmt --help
 ```
 
-Additionally, there's a wrapper called [`treefmt-nix`](https://github.com/numtide/treefmt-nix) for using `treefmt` with [`nix`](https://github.com/NixOS/nix).
+Additionally, there's a wrapper called [`treefmt-nix`](https://github.com/numtide/treefmt/v2-nix) for using `treefmt` with [`nix`](https://github.com/NixOS/nix).
```
Comment (jfly) [inline README.md:78]: Typo?

File: README.md [hunk 3]
```diff
@@ -107,12 +107,12 @@ Before specifying the formatter in the config, make sure it’s installed.
 
 To find and share existing formatter recipes, take a look at the [docs].
 
-If you are a Nix user, you might also be interested in [treefmt-nix](https://github.com/numtide/treefmt-nix) to use Nix to configure and bring in
+If you are a Nix user, you might also be interested in [treefmt-nix](https://github.com/numtide/treefmt/v2-nix) to use Nix to configure and bring in
```
Comment (jfly) [inline README.md:110]: Typo?

File: README.md [hunk 4]
```diff
@@ -182,7 +182,7 @@ Unless explicitly stated otherwise, any contribution intentionally submitted for
 
 [Brian]: https://github.com/brianmcgee
 [zimbatm]: https://github.com/zimbatm
-[Version 1]: https://github.com/numtide/treefmt/tree/v1
+[Version 1]: https://github.com/numtide/treefmt/v2/tree/v1
```
Comment (jfly) [inline README.md:185]: Typo?

File: README.md [hunk 5]
```diff
@@ -107,12 +107,12 @@ Before specifying the formatter in the config, make sure it’s installed.
...
 To find and share existing formatter recipes, take a look at the [docs].
 
-If you are a Nix user, you might also be interested in [treefmt-nix](https://github.com/numtide/treefmt-nix) to use Nix to configure and bring in
+If you are a Nix user, you might also be interested in [treefmt-nix](https://github.com/numtide/treefmt/v2-nix) to use Nix to configure and bring in
 formatters.
 
 ## Compatibility
 
-`treefmt` works with any formatter that adheres to the [following specification](https://github.com/numtide/treefmt/blob/main/docs/formatter-spec.md).
+`treefmt` works with any formatter that adheres to the [following specification](https://github.com/numtide/treefmt/v2/blob/main/docs/formatter-spec.md).
```
Comment (jfly) [inline README.md:115]: Typo?

File: docs/content/contributing/code.md [hunk 6]
```diff
@@ -59,4 +59,4 @@ For more information see the [next section](./docs.md).
 [Direnv]: https://direnv.net
 [devshell]: https://nix.dev/tutorials/first-steps/declarative-shell.html
 [treefmt]: https://treefmt.com
-[treefmt-nix]: https://github.com/numtide/treefmt-nix
+[treefmt-nix]: https://github.com/numtide/treefmt/v2-nix
```
Comment (jfly) [inline docs/content/contributing/code.md:62]: Typo?

File: docs/content/getting-started/configure.md
Comment (jfly) [inline docs/content/getting-started/configure.md:1]: I suspect nothing was supposed to change in this file.

File: docs/content/getting-started/install.md
Comment (jfly) [inline docs/content/getting-started/install.md:1]: I suspect nothing was supposed to change in this file.

File: mkdocs.yml
Comment (jfly) [inline mkdocs.yml:1]: I suspect nothing was supposed to change in this file.

File: nix/packages/treefmt/default.nix [hunk 7]
```diff
@@ -78,7 +78,7 @@ in
 
     meta = with lib; {
       description = "treefmt: one CLI to format your repo";
-      homepage = "https://github.com/numtide/treefmt";
+      homepage = "https://github.com/numtide/treefmt/v2";
```
Comment (jfly) [inline nix/packages/treefmt/default.nix:81]: Typo?

File: nix/packages/treefmt/default.nix [hunk 8]
```diff
@@ -42,8 +42,8 @@ in
     ldflags = [
       "-s"
       "-w"
-      "-X github.com/numtide/treefmt/build.Name=${pname}"
-      "-X github.com/numtide/treefmt/build.Version=v${version}"
+      "-X github.com/numtide/treefmt/v2/build.Name=${pname}"
```
Comment (jfly) [inline nix/packages/treefmt/default.nix:45]: I'm not enough of a go expert to know of these were supposed to change.

File: test/examples/nixpkgs.toml
Comment (jfly) [inline test/examples/nixpkgs.toml:1]: I suspect nothing was supposed to change in this file.

File: test/examples/treefmt.toml
Comment (jfly) [inline test/examples/treefmt.toml:1]: I suspect nothing was supposed to change in this file.

**Review by jfly — COMMENTED** [review]
I think you did an overly aggressive find/replace? I left comments inline.

I don't really understand why this change is necessary. I read [the doc] (https://go.dev/blog/v2-go-modules) linked to in the issue, and it's just talking about import compatibility. I don't see how that applies to us. We don't expose a public go api, do we?

**Comment by brianmcgee** [comment]
Yeah, it was early, I thought I was being smart. I'll have another go later. 

Whilst we don't officially support using treefmt as a library, there's nothing really stopping it either.

I think we should at least follow this go idiom, even if being used as a library is not our primary use case.

**Review by jfly — APPROVED** [review]

File: nix/packages/treefmt/default.nix (hunk 8 above)
Comment (brianmcgee) [reply nix/packages/treefmt/default.nix:45]: They were supposed to changed.

**Comment by brianmcgee** [comment]
@mergifyio queue

**Pull request merged**